packages/core/src/tracing/instrumentation/cloud/aws/dynamodb.js

```diff
@@ -0,0 +1,179 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const shimmer = require('shimmer');
+const requireHook = require('../../../../util/requireHook');
+const cls = require('../../../cls');
+const { EXIT, isExitSpan } = require('../../../constants');
+const tracingUtil = require('../../../tracingUtil');
+
+const operationsInfo = {
+  createTable: { op: 'create' },
+  // deleteTable: { op: 'delete' },
+  listTables: { op: 'list' },
+  scan: { op: 'scan' },
+  query: { op: 'query' },
+  getItem: { op: 'get' },
+  deleteItem: { op: 'delete' },
+  putItem: { op: 'put' },
+  updateItem: { op: 'update' }
+};
+
+const methodList = Object.keys(operationsInfo);
+
+let isActive = false;
+
+exports.isActive = function() {
+  return isActive;
+};
+
+exports.init = function init() {
+  requireHook.onModuleLoad('aws-sdk', instrumentAWS);
```

Comment:
We are doing that in multiple places now, that is we add multiple hooks for `'aws-sdk'`. I haven't formed an opinion on that yet but I wonder if we should only have _one_ `requireHook.onModuleLoad('aws-sdk', ...);`  call in the entire code base and branch out to the DynamoDB and S3 instrumentation from there.

Reply:
I thought about the same, but decided to move on this way for now, and to discuss into detail with you after.

Reply:
We decided to postpone this to later. The current approach is fine as it is.